Break out logging handler initialization (#4775)
cocotb/cocotb: docs/source/newsfragments/4775.removal.rst

> Removed ``cocotb.logging.SimBaseLog``. The class no longer did anything, so simply remove it.

Reviewer: ```suggestion
Removed ``cocotb.logging.SimBaseLog`` as the class no longer did anything.
```
